captum/optim/_core/output_hook.py

```diff
@@ -64,11 +39,15 @@ def forward_hook(module, input, output) -> None:
                     "As of 2019-11-22 please don't reuse nn.Modules in your models."
                 )
             if self.is_ready:
-                raise AbortForwardException("Forward hook called, all outputs saved.")
+                warn(
+                    "No outputs found from models. This can be ignored if you are "
+                    "optimizing on inputs only, without models. Otherwise, check "
+                    "that you are passing model layers in your losses."
+                )
```

Comment:
We should have tests for `NaturalImage` and transform targets, so that we don't end up breaking this functionality in the future by mistake

Reply:
Added test for NaturalImage

Reply:
Awesome! I think that we might want to also mention that transforms and NaturalImage can be used as targets in the tutorial notebooks somewhere.